.github/workflows/build.yml

```diff
@@ -13,3 +13,25 @@ jobs:
       target-name: 'dd_pprof' # target name in binding.gyp
       package-manager: 'npm' # npm or yarn
       cache: true # enable caching of dependencies based on lockfile
+
+  publish:
+    if: github.ref_name == 'v1.x'
```

Comment:
Is there some way we can make this generic? Don't want this blowing up the next time we do a major. 😞

Reply:
Actually, I don't think this will even work in build.yml because it will trigger on all pull requests too, which should not happen, it should only trigger on pushes to the v1.x branch. But even with that ref_name guard it will never happen because build.yml only triggers on pushes to main.

Reply:
Actually...doesn't it need something to download the artifacts too?

Reply:
> Is there some way we can make this generic? Don't want this blowing up the next time we do a major.

The idea was that we simply bump this for majors, but if you have a better idea to not have to change this it would definitely be best.

> Actually, I don't think this will even work in build.yml because it will trigger on all pull requests too

Ideally this wouldn't happen, but it might be acceptable for now since the deployment needs to be manually approved in the UI. Maybe we could add another condition to avoid PRs?

> Actually...doesn't it need something to download the artifacts too?

Yes, forgot about it, adding now.

Reply:
Given how simple the build part of the script is, perhaps it would be better to move the release stuff back to a separate script and just copy it in _that_ script? Then we could restore the `v[0-9]+.x` bit and not have to worry about majors _or_ this triggering on unrelated branches?

Reply:
But then you would have to list the branches that shouldn't run the other script to avoid both running. It might still be easier though, not sure.

Reply:
Not sure what you mean. The other script would just have this like it used to:

```yml
on:
  push:
    branches:
      - v[0-9]+.x
```

That'd run only on pushes to the release branches, and this build.yml already has:

```yml
on:
  pull_request:
  push:
    branches:
      - main
```

So it will only run on `main` and pull requests.

Reply:
I added a commit restoring the workflow split but with the proper build step and download-artifacts action.